Firestore/core/src/firebase/firestore/util/type_traits.h

```diff
@@ -45,43 +47,25 @@ namespace util {
  * is_objective_c_pointer<std::unique_ptr<int>>::value == false
  */
 template <typename T>
-struct is_objective_c_pointer {
- private:
-  using yes_type = char (&)[10];
-  using no_type = char (&)[1];
-
-  /**
-   * A non-existent function declared to produce a pointer to type T (which is
-   * consistent with the way Objective-C objects are referenced).
-   *
-   * Note that there is no definition for this function but that's okay because
-   * we only need it to reason about the function's type at compile type.
-   */
-  static T Pointer();
-
-  static yes_type Choose(id value);
-  static no_type Choose(...);
-
- public:
-  using value_type = bool;
-
-  enum { value = sizeof(Choose(Pointer())) == sizeof(yes_type) };
+struct is_objective_c_pointer : std::is_convertible<T, id> {};
```

Comment:
I _think_ this is equivalent to the previous implementation (and the tests pass), but I'll run it by Gil before merging.

Reply:
If `TEST(TypeTraitsTest, IsObjectiveCPointer)` passes I'm all for having less code.

Could this be a using statement? Something like `using is_objective_c_pointer = std::is_convertible<T, id>`?

Reply:
Done, good idea.